### PR TITLE
Add domain glob to feeds

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
- [ ] In Progress
- [ ] Ready
- [x] In Hold - (Reason for hold)

## Description
Added domainglob new indicator type to Bambenek, phishme, proofpoint, recordedfuture and o365

## Related Issue
fixes: https://github.com/demisto/etc/issues/22118

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [x] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation 

## Office 365 Test playbook
Added the 6 last task
Office365_Feed_Test_Tue_Mar_03_2020

### CODE REVIEW

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py [hunk 1]
```diff
@@ -164,6 +172,7 @@ def module_test_command(client: Client, args):
                 FeedIndicatorType.Account,
                 FeedIndicatorType.CVE,
                 FeedIndicatorType.Domain,
+                FeedIndicatorType.DomainGlob,
                 FeedIndicatorType.Email,
                 FeedIndicatorType.File,
                 FeedIndicatorType.MD5,
```
Comment (avidan-H) [inline Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: Why are we maintaining a list of the supported values here? Let's make it an attribute on the `FeedIndicatorType` class itself.

File: Packs/FeedOffice365/Integrations/FeedOffice365/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ## [Unreleased]
-
+Added new indicator type **DomainGlobe**
```
Comment (avidan-H) [inline Packs/FeedOffice365/Integrations/FeedOffice365/CHANGELOG.md]:
```suggestion
Added **DomainGlob** to the integration's indicator types.
```

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.yml [hunk 3]
```diff
@@ -111,6 +111,7 @@ script:
       - all
       - domain
       - ip
+      - domainGlob
```
Comment (avidan-H) [inline Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.yml]: Let's not add this option.

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py [hunk 4]
```diff
@@ -87,6 +87,10 @@ def _build_iterator(
                 # add type/value to item.
                 if "domain" in item:
                     item["type"] = FeedIndicatorType.Domain
+                    # As part of the domain feed, also DomainGlob indicators will be returned, so we are checking if the
+                    # domain has '*' in their value
+                    if '*' in str(item.get("domain")):
+                        item["type"] = FeedIndicatorType.DomainGlob
                     item["value"] = item.get("domain")
```
Comment (avidan-H) [inline Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py]:
```suggestion
                    ioc_value = item.get("domain", "")
                    if '*' in ioc_value:
                        item["type"] = FeedIndicatorType.DomainGlob
                    item["value"] = ioc_value
```
Shouldn't the value always be a string? I don't think there is a need to explicitly cast to a `str`.

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/CHANGELOG.md [hunk 5]
```diff
@@ -1,2 +1,3 @@
 ## [Unreleased]
+Added **DomainGlob** to the feed's indicators.
```
Comment (avidan-H) [inline Packs/FeedProofpoint/Integrations/FeedProofpoint/CHANGELOG.md]:
```suggestion
Added **DomainGlob** to the integration's indicator types.
```

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint_test.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 from FeedProofpoint import Client, fetch_indicators_command
+from CommonServerPython import *
```
Comment (avidan-H) [inline Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint_test.py]:
```suggestion
from CommonServerPython import FeedIndicatorType
```

File: Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py [hunk 7]
```diff
@@ -225,6 +225,9 @@ def get_indicator_type(indicator_type, item):
     elif indicator_type == 'hash':
         return FeedIndicatorType.File
     elif indicator_type == 'domain':
+        # If * is in the domain it is of type domain globe
```
Comment (avidan-H) [inline Packs/FeedRecordedFuture/Integrations/FeedRecordedFuture/FeedRecordedFuture.py]:
```suggestion
        # If * is in the domain it is of type DomainGlob
```

File: Packs/FeedCofense/Integrations/FeedCofense/FeedCofense.py [hunk 8]
```diff
@@ -132,6 +132,9 @@ def _convert_block(cls, block: dict) -> Tuple[str, str]:
             value = block.get("data_1", {}).get("url")
         else:
             value = block.get("data_1")
+            # If a domain has '*' in the value it is of type domain globe
```
Comment (avidan-H) [inline Packs/FeedCofense/Integrations/FeedCofense/FeedCofense.py]:
```suggestion
            # If a domain has '*' in the value it is of type DomainGlob
```

**Comment by avidan-H** [comment]
Post pictures of the test playbooks you modified as well, please.

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py (hunk 1 above)
Comment (ShahafBenYakir) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: Not my decision nor implementation I assume that the reason is that this is a subset of indicators, not all from CSP.

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py (hunk 4 above)
Comment (ShahafBenYakir) [reply Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py]: The str cast is needed it sometimes fails in unit tests

File: Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py (hunk 1 above)
Comment (avidan-H) [reply Packs/ApiModules/Scripts/CSVFeedApiModule/CSVFeedApiModule.py]: This is the CSV module. It's meant to be used for specific implementations so by definition it has to be broad - aka it is all types and if it isn't, it should be. This should be moved to CSP.

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py (hunk 4 above)
Comment (avidan-H) [reply Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py]: Sababushka - so leave that in. You can still implement the rest of the suggested change.

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.yml [hunk 9]
```diff
@@ -56,7 +56,7 @@ configuration:
 - defaultvalue: "60"
   display: Feed Fetch Interval
   name: feedFetchInterval
-  required: true
+  required: false
```
Comment (ronykoz) [inline Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.yml]: why is this false?

**Comment by ShahafBenYakir** [comment]
@avidan-H Please approve 🙏

**Comment by Itay4** [comment]
@kirbles19 please review

File: Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py [hunk 10]
```diff
@@ -87,7 +87,12 @@ def _build_iterator(
                 # add type/value to item.
                 if "domain" in item:
                     item["type"] = FeedIndicatorType.Domain
-                    item["value"] = item.get("domain")
+                    # As part of the domain feed, also DomainGlob indicators will be returned, so we are checking if the
+                    # domain has '*' in their value
+                    indicator_value = str(item.get("domain", ""))
```
Comment (avidan-H) [inline Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py]: Why are you calling `str` here?

Comment (ShahafBenYakir) [reply Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py]: For some reason running line 93 - `if '*' in indicator_value:` resulted in a type error was solved by adding this casting. I Could change the str casting to be inside the if statement like this: `if '*' in str(indicator_value):`

Comment (avidan-H) [reply Packs/FeedProofpoint/Integrations/FeedProofpoint/FeedProofpoint.py]: I suppose that could happen if the value for that key is explicitly `None` but if that were the case, then we shouldn't even be yielding an indicator object for that iteration. Please investigate why that may be happening. If there are iterations for which the key `domain` is present but the value is `None`, we should be filtering those out.

**Comment by kirbles19** [comment]
@ShahafBenYakir Done.

**Comment by avidan-H** [comment]
You need to pull from `master`. Your branch isn't up to date since the release went out yesterday.

**Comment by ShahafBenYakir** [comment]
@avidan-H Yep, saw it now and pulled. Should be fine now.

**Comment by ShahafBenYakir** [comment]
@glicht / @anara123 This is the PR for adding DomainGlob indicator type to the following feeds:
- Bambenek
- Phishme
- Proofpoint
- RecordedFuture 
- Office365

Due to small addition to `CommonServerPython` all test playbooks are running (and obviously some fail) thus I need force merge. Unit tests linting and all other steps pass successfully.

**Pull request merged**